disnake/ext/commands/params.py

```diff
@@ -1115,10 +1242,24 @@ def converter_method(function: Any) -> ConverterMethod:
         return ConverterMethod(function)
 
 
-def register_injection(function: CallableT) -> CallableT:
+def register_injection(
+    function: Callable, *, autocompleters: Dict[str, Callable] = ...
+) -> Injection:
     """A decorator to register a global injection.
 
     .. versionadded:: 2.3
+
+    .. versionchanged:: 2.6
+        Now returns :class:`disnake.ext.commands.Injection`
+
+    Raises
+    ------
+    TypeError
+        Injection don't have a return annotation
+
+        Injection cannot overwrite builtin types
+
+        Any of autocompleters is not a coroutine
```

Comment:
```suggestion
        Injection doesn't have a return annotation,
        or tries to overwrite builtin types.
```

Reply:
Done in [9136fcf412f032a0268c589a541efdcd819844fb](https://github.com/DisnakeDev/disnake/pull/675/commits/9136fcf412f032a0268c589a541efdcd819844fb) (i lost connection and was not able to merge it, but had a screenshot in local "TODO")